src/data_access/FileUserDataAccessObject.java

```diff
@@ -0,0 +1,92 @@
+package data_access;
+
+import entity.*;
+
+import java.io.*;
+import java.util.*;
+
+public class FileUserDataAccessObject {
+    private final File usersFile;
+
+    // contains the content in each column
+    private final Map<String, Integer> headers = new LinkedHashMap<>();
+
+    // mapping username to User
+    private final Map<String, User> accounts = new HashMap<>();
+
+    private UserFactory userFactory;
+
+    public FileUserDataAccessObject(String csvPath, UserFactory userFactory) throws IOException {
+        this.userFactory = userFactory;
+        this.usersFile = new File(csvPath);
+        headers.put("username", 0);
+        headers.put("password", 1);
+        headers.put("bio", 2);
+        headers.put("artists", 3);
+        headers.put("spotify_handle", 4);
+        headers.put("playlistId", 5);
+        headers.put("friends", 6);
+        headers.put("requests", 7);
+
+        if (this.usersFile.length() == 0) {
+            save();
+        } else {
+
+            try (BufferedReader reader = new BufferedReader(new FileReader(usersFile))) {
+
+                String row;
+                while ((row = reader.readLine()) != null) {
+                    String[] col = row.split(",");
```

Comment:
I think we should use a different delimiter instead of commas for the csv because one column can contain several commas in itself. For example, a user's bio may have multiple commas.

Reply:
That's right! Should we use double quotes as the delimiter for each column (username, password, bio, etc) and commas within the ones that are ArrayLists (friends and artists)?

Reply:
Yes, I believe that works. I think we can also use pipe (|) as the delimiter.